tests/rbd/test_rbd_image_operations.py

```diff
@@ -0,0 +1,295 @@
+
+        log.info(f"Rename images in pool {pool} and verify")
+        image_rename_config = {
+            k: {"rename_to": k + "_renamed"} for k in multi_image_config.keys()
+        }
+        rc = wrapper_for_image_ops(
+            rbd=rbd,
+            pool=pool,
+            image_config=image_rename_config,
+            ops_module="ceph.rbd.workflows.rbd",
+            ops_method="rename_single_image_and_verify",
+            test_ops_parallely=test_ops_parallely,
+        )
+        if rc:
+            log.error(f"Rename image and verify failed for pool {pool}")
+            return 1
+        log.info(f"Rename image and verify success for pool {pool}")
+
+        image_keys = list(multi_image_config.keys())
+        for image in image_keys:
+            new_name = image_rename_config[image]["rename_to"]
+            multi_image_config[new_name] = multi_image_config.pop(image)
+
+        new_pool = pool + "_new"
+        log.info(f"Create pool {new_pool} to perform pool copy operation")
+
+        config = kw.get("config", {})
+        is_ec_pool = True if "ec" in pool_type else False
+        new_pool_config = deepcopy(pool_config)
+        if is_ec_pool:
+            data_pool_new = pool_config["data_pool"] + "_new"
+            new_pool_config["data_pool"] = data_pool_new
+        rc = create_single_pool_and_images(
+            config=config,
+            pool=new_pool,
+            pool_config=new_pool_config,
+            client=client,
+            cluster="ceph",
+            rbd=rbd,
+            ceph_version=int(config.get("rhbuild")[0]),
+            is_ec_pool=is_ec_pool,
+            is_secondary=False,
+            do_not_create_image=True,
+        )
+        if rc:
+            log.error(f"Creation of pool {new_pool} failed")
+            return rc
+
+        log.info(f"Copy images from pool {pool} to another pool {new_pool} and verify")
+        rc = wrapper_for_image_ops(
+            rbd=rbd,
+            pool=pool,
+            image_config=multi_image_config,
+            new_pool=new_pool,
+            ops_module="ceph.rbd.workflows.rbd",
+            ops_method="copy_single_image_and_verify",
+            test_ops_parallely=test_ops_parallely,
+        )
+        if rc:
+            log.error(f"Copy images from pool {pool} failed")
+            return 1
+        log.info(f"Copy images from pool {pool} success")
+
+        exec_cmd(node=client, cmd="ceph config set mon mon_allow_pool_delete true")
+        sleep(20)
+
+        rc = exec_cmd(
+            cmd=f"ceph osd pool delete {new_pool} {new_pool} "
+            "--yes-i-really-really-mean-it",
+            node=client,
+        )
+        if rc:
+            log.error(f"Error while deleting new pool {new_pool}")
+            if test_ops_parallely:
+                raise Exception(f"Error while deleting new pool {new_pool}")
+            return 1
+
+        if is_ec_pool:
+            rc = exec_cmd(
```

Comment:
can we create one pool deletion method instead of adding same command to delete multiple pool?

Reply:
it's just one function exec_cmd call right. I don't think further customisation is necessary here.